test/e2e/util.go

```diff
@@ -189,3 +194,43 @@ func debugJSON(v interface{}) string {
 	raw, _ := json.MarshalIndent(v, "", "  ")
 	return string(raw)
 }
+
+// SetupRbacInNamespace deploys appropriate role and its binding to service
+// account in given namespace for the test to be able to run seccessfuly
+func SetupRbacInNamespace(namespace string) error {
+	// get eunomia root dir
+	_, utilTestFilename, _, _ := runtime.Caller(0)
+	eunomiaRoot := filepath.Join(filepath.Dir(utilTestFilename), "../..")
+
+	// create temporary file to store kubectl input in
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "sa-")
+	if err != nil {
+		return fmt.Errorf("Failed to create temporary file")
+	}
+	defer os.Remove(tmpFile.Name())
+
+	// create kubectl input
+	out, err := exec.Command(
+		"helm", "template",
+		filepath.Join(eunomiaRoot, "deploy/test/eunomia-rbac/"),
+		"--set", "eunomia.namespace="+namespace,
+	).Output()
```

Comment:
There is the [InitializeClusterResources() function](https://github.com/operator-framework/operator-sdk/blob/v0.8.1/pkg/test/resource_creator.go#L108) using namespacedMan parameter I didn't see earlier - I'll try to come up with something using it (but I will also need a namespaced manifest dir - maybe I'll be generating it dynamically somehow).

Reply:
The function I linked to isn't enough unfortunately. I can create a single manifest file like this one:
<details><summary><code>my_manifest.yml</code></summary><br><pre>
---
# Source: eunomia-rbac/templates/service-account-operator.yaml
# service account for the operator
apiVersion: v1
kind: ServiceAccount
metadata:
  name: eunomia-operator
  namespace: eunomia-operator
---
# Source: eunomia-rbac/templates/role-operator.yaml
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: eunomia-operator
  namespace: "eunomia-operator"
rules:
# leader election   
- apiGroups:
  - ""
  resources:
  - configmaps
  - pods
  verbs:
  - "*"  
#Metrics  
- apiGroups:
  - ""
  resources:
  - services
  verbs:
  - "*"
#Metrics  
- apiGroups:
  - "apps"
  resources:
  - replicasets
  - deployments
  verbs: 
  - "*"
# In Kubernetes <1.16, "deployments" also require API group "extensions" (see: https://stackoverflow.com/a/57854939)
- apiGroups:
  - "extensions"
  resources:
  - deployments
  verbs:
  - "*"
# Metrics  
- apiGroups:
  - monitoring.coreos.com
  resources:
  - servicemonitors
  verbs:
  - "get"
  - "create"
# Metrics  
- apiGroups:
  - apps
  resources:
  - deployments/finalizers
  resourceNames:
  - eunomia-operator
  verbs:
  - "update"
---
# Source: eunomia-rbac/templates/service-account-operator.yaml
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: eunomia-operator
  namespace: eunomia-operator
subjects: 
- kind: ServiceAccount
  name: eunomia-operator
  namespace: eunomia-operator
roleRef:
  kind: Role
  name: eunomia-operator
  apiGroup: rbac.authorization.k8s.io
</pre></details>

and pass it to operator-sdk: (`operator-sdk test local ./test/e2e --namespaced-manifest /path/to/my_manifest.yml`.

But this is what happens:
In each test function I invoke:
- [`InitializeClusterResources`](https://github.com/operator-framework/operator-sdk/blob/v0.8.1/pkg/test/resource_creator.go#L108),
- which returns [`ctx.createFromYAML`](https://github.com/operator-framework/operator-sdk/blob/v0.8.1/pkg/test/resource_creator.go#L55),
- this function [sets the current namespace](https://github.com/operator-framework/operator-sdk/blob/v0.8.1/pkg/test/resource_creator.go#L72) in the resources provided by my `my_manifest.yml`,
- but unfortunately only in the [metadata section](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/unstructured.go#L237-L243).

This is a problem for me because I need the namespace to be also set in `my_manifest.yml` in RoleBinding's subjects subsection. It doesn't happen and the namespace there stays set to the initial value (it isn't being changed per test to the dynamically created namespace as in the metadata section) so I get access errors.

Reply:
After additional research I found out that if you don't pass `--namespaced-manifest` flag with content to the `operator-sdk test local` command (and at the same time you don't provide `--no-setup` flag) the program looks for [some default files](https://github.com/operator-framework/operator-sdk/blob/v0.8.1/cmd/operator-sdk/test/local.go#L122) which we do not have in `deploy/` directory (maybe they were created by the `operator-sdk` initially but then removed at some point in time because we started using helm?).